/* Copyright 2019 Alexander Belinsky. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at
    http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
==============================================================================*/

syntax = "proto3";

package unetwork;

import "unetwork/core/optimization_param.proto";

// Protocol buffer representing action in the environment.
// Action corresponds to changing the state of the controlled parameters' set.
// Can be concerned as `SET parameters' values`
message Action {
    // Controlled parameters list.  
    repeated OptimizationParameter optimization_params = 1;

    // Additional info which can be associated with the parameter. 
    // Useful for debug options.
    string info = 2;
}

